File: ltexls-core/src/test/java/org/bsplines/ltexls/parsing/markdown/LtexMarkdownDisplayMathTest.java
/* Copyright (C) 2020 Julian Valentin, LTeX Development Community
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 */

package org.bsplines.ltexls.parsing.markdown;

import com.vladsch.flexmark.ast.Paragraph;
import com.vladsch.flexmark.util.ast.BlockContent;
import com.vladsch.flexmark.util.sequence.BasedSequence;
import java.util.Arrays;
import java.util.Collections;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

public class LtexMarkdownDisplayMathTest {
  @Test
  public void testConstructors() {
    LtexMarkdownDisplayMath node = new LtexMarkdownDisplayMath();
    Assertions.assertTrue(node.getOpeningMarker().isNull());
    Assertions.assertTrue(node.getOpeningTrailing().isNull());
    Assertions.assertTrue(node.getClosingMarker().isNull());
    Assertions.assertTrue(node.getClosingTrailing().isNull());

    node = new LtexMarkdownDisplayMath(BasedSequence.NULL);
    Assertions.assertTrue(node.getOpeningMarker().isNull());
    Assertions.assertTrue(node.getOpeningTrailing().isNull());
    Assertions.assertTrue(node.getClosingMarker().isNull());
    Assertions.assertTrue(node.getClosingTrailing().isNull());

    node = new LtexMarkdownDisplayMath(BasedSequence.NULL, Collections.emptyList());
    Assertions.assertTrue(node.getOpeningMarker().isNull());
    Assertions.assertTrue(node.getOpeningTrailing().isNull());
    Assertions.assertTrue(node.getClosingMarker().isNull());
    Assertions.assertTrue(node.getClosingTrailing().isNull());

    node = new LtexMarkdownDisplayMath(new BlockContent());
    Assertions.assertTrue(node.getOpeningMarker().isNull());
    Assertions.assertTrue(node.getOpeningTrailing().isNull());
    Assertions.assertTrue(node.getClosingMarker().isNull());
    Assertions.assertTrue(node.getClosingTrailing().isNull());
  }

  @Test
  public void testGetSegments() {
    LtexMarkdownDisplayMath node = new LtexMarkdownDisplayMath();
    node.setOpeningMarker(BasedSequence.of("$$"));
    node.setOpeningTrailing(BasedSequence.of("abc"));
    node.setClosingMarker(BasedSequence.of("$$"));
    node.setClosingTrailing(BasedSequence.of("def"));
    Assertions.assertEquals(Arrays.asList(BasedSequence.of("$$"), BasedSequence.of("abc"),
          BasedSequence.of("$$"), BasedSequence.of("def")),
        Arrays.asList(node.getSegments()));
  }

  @Test
  public void testGetAstExtra() {
    LtexMarkdownDisplayMath node = new LtexMarkdownDisplayMath();
    node.setOpeningMarker(BasedSequence.of("$$"));
    node.setOpeningTrailing(BasedSequence.of("abc"));
    node.setClosingMarker(BasedSequence.of("$$"));
    node.setClosingTrailing(BasedSequence.of("def"));

    StringBuilder stringBuilder = new StringBuilder();
    node.getAstExtra(stringBuilder);

    Assertions.assertEquals(" open:[0, 2, \"$$\"] openTrail:[0, 3, \"abc\"] "
        + "close:[0, 2, \"$$\"] closeTrail:[0, 3, \"def\"]",
        stringBuilder.toString());
  }

  @Test
  public void testIsParagraphEndWrappingDisabled() {
    LtexMarkdownDisplayMath node = new LtexMarkdownDisplayMath();
    Paragraph paragraph = new Paragraph();
    Assertions.assertFalse(node.isParagraphEndWrappingDisabled(paragraph));
  }

  @Test
  public void testIsParagraphStartWrappingDisabled() {
    LtexMarkdownDisplayMath node = new LtexMarkdownDisplayMath();
    Paragraph paragraph = new Paragraph();
    Assertions.assertFalse(node.isParagraphStartWrappingDisabled(paragraph));
  }

  @Test
  public void testProperties() {
    LtexMarkdownDisplayMath node = new LtexMarkdownDisplayMath();
    node.setOpeningMarker(BasedSequence.of("$$"));
    Assertions.assertEquals(BasedSequence.of("$$"), node.getOpeningMarker());

    node = new LtexMarkdownDisplayMath();
    node.setOpeningTrailing(BasedSequence.of("abc"));
    Assertions.assertEquals(BasedSequence.of("abc"), node.getOpeningTrailing());

    node = new LtexMarkdownDisplayMath();
    node.setClosingMarker(BasedSequence.of("$$"));
    Assertions.assertEquals(BasedSequence.of("$$"), node.getClosingMarker());

    node = new LtexMarkdownDisplayMath();
    node.setClosingTrailing(BasedSequence.of("def"));
    Assertions.assertEquals(BasedSequence.of("def"), node.getClosingTrailing());
  }
}

File: src/test/kotlin/org/bsplines/ltexls/parsing/markdown/LtexMarkdownDisplayMathTest.kt
/* Copyright (C) 2019-2021 Julian Valentin, LTeX Development Community
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 */

package org.bsplines.ltexls.parsing.markdown

import com.vladsch.flexmark.ast.Paragraph
import com.vladsch.flexmark.util.ast.BlockContent
import com.vladsch.flexmark.util.sequence.BasedSequence
import org.junit.platform.suite.api.IncludeEngines
import kotlin.test.Test
import kotlin.test.assertEquals
import kotlin.test.assertFalse
import kotlin.test.assertTrue

@IncludeEngines("junit-jupiter")
class LtexMarkdownDisplayMathTest {
  @Test
  fun testConstructors() {
    var node = LtexMarkdownDisplayMath()
    assertTrue(node.openingMarker.isNull)
    assertTrue(node.openingTrailing.isNull)
    assertTrue(node.closingMarker.isNull)
    assertTrue(node.closingTrailing.isNull)

    node = LtexMarkdownDisplayMath(BasedSequence.NULL)
    assertTrue(node.openingMarker.isNull)
    assertTrue(node.openingTrailing.isNull)
    assertTrue(node.closingMarker.isNull)
    assertTrue(node.closingTrailing.isNull)

    node = LtexMarkdownDisplayMath(BasedSequence.NULL, emptyList())
    assertTrue(node.openingMarker.isNull)
    assertTrue(node.openingTrailing.isNull)
    assertTrue(node.closingMarker.isNull)
    assertTrue(node.closingTrailing.isNull)

    node = LtexMarkdownDisplayMath(BlockContent())
    assertTrue(node.openingMarker.isNull)
    assertTrue(node.openingTrailing.isNull)
    assertTrue(node.closingMarker.isNull)
    assertTrue(node.closingTrailing.isNull)
  }

  @Test
  fun testGetSegments() {
    val node = LtexMarkdownDisplayMath()

    node.openingMarker = BasedSequence.of("$$")
    node.openingTrailing = BasedSequence.of("abc")
    node.closingMarker = BasedSequence.of("$$")
    node.closingTrailing = BasedSequence.of("def")

    assertEquals(
      listOf(
        BasedSequence.of("$$"),
        BasedSequence.of("abc"),
        BasedSequence.of("$$"),
        BasedSequence.of("def"),
      ),
      node.segments.asList(),
    )
  }

  @Test
  fun testGetAstExtra() {
    val node = LtexMarkdownDisplayMath()
    node.openingMarker = BasedSequence.of("$$")
    node.openingTrailing = BasedSequence.of("abc")
    node.closingMarker = BasedSequence.of("$$")
    node.closingTrailing = BasedSequence.of("def")

    val stringBuilder = StringBuilder()
    node.getAstExtra(stringBuilder)

    assertEquals(
      " open:[0, 2, \"$$\"] openTrail:[0, 3, \"abc\"] "
          + "close:[0, 2, \"$$\"] closeTrail:[0, 3, \"def\"]",
      stringBuilder.toString()
    )
  }

  @Test
  fun testIsParagraphEndWrappingDisabled() {
    val node = LtexMarkdownDisplayMath()
    val paragraph = Paragraph()
    assertFalse(node.isParagraphEndWrappingDisabled(paragraph))
  }

  @Test
  fun testIsParagraphStartWrappingDisabled() {
    val node = LtexMarkdownDisplayMath()
    val paragraph = Paragraph()
    assertFalse(node.isParagraphStartWrappingDisabled(paragraph))
  }

  @Test
  fun testProperties() {
    var node = LtexMarkdownDisplayMath()
    node.openingMarker = BasedSequence.of("$$")
    assertEquals(BasedSequence.of("$$"), node.openingMarker)

    node = LtexMarkdownDisplayMath()
    node.openingTrailing = BasedSequence.of("abc")
    assertEquals(BasedSequence.of("abc"), node.openingTrailing)

    node = LtexMarkdownDisplayMath()
    node.closingMarker = BasedSequence.of("$$")
    assertEquals(BasedSequence.of("$$"), node.closingMarker)

    node = LtexMarkdownDisplayMath()
    node.closingTrailing = BasedSequence.of("def")
    assertEquals(BasedSequence.of("def"), node.closingTrailing)
  }
}
